app/models/GroupMembership.ts

```diff
@@ -47,6 +51,31 @@ class GroupMembership extends NavigableModel {
 
   // methods
 
+  /**
+   * Adds the document identified by the given id to the group membership in
+   * memory. Does not add the document to the database or store.
+   *
+   * @param document The document to add.
+   * @param parentDocumentId The id of the document to add the new document to.
+   */
+  @action
+  addDocument(document: Document, parentDocumentId?: string) {
```

Comment:
Seems good but both of these inherit from `NavigableModel` so we could probably just put it there once